streams/src/test/java/org/apache/kafka/streams/integration/EosBetaUpgradeIntegrationTest.java

```diff
@@ -906,29 +912,22 @@ public void close() { }
         return streams;
     }
 
-    private void waitForStateTransition(final List<KeyValue<KafkaStreams.State, KafkaStreams.State>> observed,
-                                        final List<KeyValue<KafkaStreams.State, KafkaStreams.State>> expected)
-            throws Exception {
-
+    private void waitForRunning(final List<KeyValue<KafkaStreams.State, KafkaStreams.State>> observed) throws Exception {
         waitForCondition(
-            () -> observed.equals(expected),
+            () -> observed.get(observed.size() - 1).equals(new KeyValue<>(State.REBALANCING, State.RUNNING)),
```

Comment:
Seems this could fail if `observed` is empty?

Also, could we simplify this? We are only interested in `RUNNING ` state, right? So no need to check the previous state?

Reply:
True, I'll fix this